pinot-common/src/main/java/com/linkedin/pinot/common/segment/crypt/PinotCrypterFactory.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.common.segment.crypt;
+
+import org.apache.commons.configuration.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * This factory instantiates the PinotCrypter, which participates in encrypting and decrypting files.
+ */
+public class PinotCrypterFactory {
+  private static final String CRYPTER_CLASS_CONFIG = "class";
+  private static String _crypterClass;
+  public static final Logger LOGGER = LoggerFactory.getLogger(PinotCrypterFactory.class);
+
+  // Prevent factory from being instantiated
+  private PinotCrypterFactory() {
+
+  }
+  public static void init(Configuration config) {
+    _crypterClass = config.getString(CRYPTER_CLASS_CONFIG);
+  }
+
+  public static PinotCrypter create() {
```

Comment:
Create() in a factory should take an argument on what concrete implementation to instantiate, otherwise it is not a factory pattern.

Reply:
We will instantiate the PinotCrypter class at time of starting controllers, servers, minions so that when we download segments, we can decrypt. This class will be wired in by reflection, and during init, we will take that parameter in, so we don't need another parameter during create.